backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/pporder/api/impl/HUPPOrderBL.java

```diff
@@ -81,9 +82,9 @@ public class HUPPOrderBL implements IHUPPOrderBL
 	private final IHUAttributesBL attributesBL = Services.get(IHUAttributesBL.class);
 	private final IProductPlanningDAO productPlanningDAO = Services.get(IProductPlanningDAO.class);
 
-	private final HUQRCodesService huqrCodesService = SpringContextHolder.instance.getBean(HUQRCodesService.class);
-	private final MaturingConfigRepository maturingConfigRepository = SpringContextHolder.instance.getBean(MaturingConfigRepository.class);
-	private final PPOrderCandidateDAO ppOrderCandidateDAO = SpringContextHolder.instance.getBean(PPOrderCandidateDAO.class);
+	private final ExtendedMemorizingSupplier<HUQRCodesService> huqrCodesServiceSupplier = ExtendedMemorizingSupplier.of(() -> SpringContextHolder.instance.getBean(HUQRCodesService.class));
```

Comment:
i think u are using ExtendedMemorizingSupplier because you want to make those bean instances lazy.
If that's the case, pls use org.compiere.SpringContextHolder#lazyBean(java.lang.Class<T>)